node/overseer/src/lib.rs

```diff
@@ -330,6 +331,38 @@ pub async fn forward_events<P: BlockchainEvents<Block>>(client: Arc<P>, mut hand
 	}
 }
 
+/// Used to detect if the node is in major sync. This can happen only on startup so once syncing is
```

Comment:
That is not really true, you can also fall back to major sync mode at runtime. You may loose all peers or have some connectivity problems etc.

Reply:
My wording is bad here. I meant that we treat only the initial sync specially. Even if major sync happens afterwards subsystems shouldn't be blocked, or not participate in consensus, etc.

I'll fix this, but generally the whole PR is at quick and dirty test phase. I want to see it works fine on Versi before doing it properly.

Reply:
Okay :P

Reply:
But be aware that subsystem aren't informed about block imports on major sync. So, they probably don't do anything.

Reply:
Which is actually good. But the problem is that an 'initial' `ActiveLeavesUpdate` is sent before sync and a lot of subsystems start doing work (and generate errors if they need the runtime).

Do I understand correctly that the network client doesn't generate `BlockImported` during major sync? This makes some of my changes pointless.

Reply:
You mean block import notifications? No there shouldn't be any notifications coming. But there are some cases when you can receive a block import notification on major sync, for example when there is a reorg.

Reply:
> You mean block import notifications? No there shouldn't be any notifications coming. But there are some cases when you can receive a block import notification on major sync, for example when there is a reorg.

@bkchr can you explain in which cases we will receive import notifications?
If we really don't get any notifications for old finalized data, we might just remove the initial ActiveLeavesUpdate as @eskimor suggests and make this PR one liner :)

Reply:
You don't get block import notifications for blocks that you already have imported. When blocks are finalized there will be a finalized notification. 



> If we really don't get any notifications for old finalized data,

What exactly do you mean by this?

Reply:
I was asking about import notifications during (initial) major sync. I think you said they are sent only on special cases (reorg was one of them). I'm asking about these special cases.

> What exactly do you mean by this?

Let's say the node is syncing. Is there a case when we get import notification for data which is already finalized?

Reply:
> Let's say the node is syncing. Is there a case when we get import notification for data which is already finalized?

No

> I was asking about import notifications during (initial) major sync. I think you said they are sent only on special cases (reorg was one of them). I'm asking about these special cases.

There aren't more and re-org on major sync only happens if the node was once synced and then was restarted.

Reply:
Got it, thank you one more time!